pkg/ccl/importccl/exportcsv.go

```diff
@@ -131,6 +133,20 @@ func newCSVWriterProcessor(
 	input execinfra.RowSource,
 	output execinfra.RowReceiver,
 ) (execinfra.Processor, error) {
+	// Check if the user setting up the processor has sufficient privileges to access the
+	// ExternalStorage destination.
+	if !spec.IsAdmin {
```

Comment:
This -- the last processor during _execution_ -- feels very late to be doing authorization checks doesn't it?